### PR TITLE
Modify the docstring of qml.eigvals

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
This PR modifies the docstring of `qml.eigvals`.

**Description of the Change:**
The backticks are corrected and the method options for computing the eigenvalues are added to the Args.

**Benefits:**
Better readability of the function docstring.

### CODE REVIEW

**Review by rmoyard — COMMENTED** [review]
Thanks for the changes @soranjh, I've one question. When it is resolve, I am happy to approve it 👍

File: pennylane/ops/functions/eigvals.py [hunk 1]
```diff
@@ -29,20 +29,21 @@ def eigvals(op, k=1, which="SA"):
...
-    ``scipy.sparse.linalg.eigsh`` fails. If the requested `k` is equal or larger than `N - 1`, the
-    regular ``qml.math.linalg.eigvalsh`` is applied on the dense matrix. The possible methods for
-    computing the `k` eigenvalues are `"LM"` (largest in magnitude), `"SM"` (smallest in magnitude),
-    `"LA"` (largest algebraic), `"SA"` (smallest algebraic) and `"BE"` (k/2 from each end of the
-    spectrum). For more details see
+    ``scipy.sparse.linalg.eigsh`` method which returns :math:`k` eigenvalues. The default value of
+    :math:`k` is :math:`1`. For an :math:`N \times N` sparse matrix, :math:`k` must be smaller than
+    :math:`N - 1`, otherwise ``scipy.sparse.linalg.eigsh`` fails. If the requested :math:`k` is
+    equal or larger than :math:`N - 1`, the regular ``qml.math.linalg.eigvalsh`` is applied on the
+    dense matrix. For more details see
```
[the remark this answers is not in the document]
Comment (soranjh) [reply pennylane/ops/functions/eigvals.py:36]: This sentence?

> If the requested :math:`k` is equal or larger than :math:`N - 1`, the regular ``qml.math.linalg.eigvalsh`` is applied on the dense matrix.

[the remark this answers is not in the document]
Comment (rmoyard) [reply pennylane/ops/functions/eigvals.py]: I was thinking about the whole paragraph that is about `SparseHamiltonian`

[the remark this answers is not in the document]
Comment (soranjh) [reply pennylane/ops/functions/eigvals.py]: Sure, the paragraph is a note now.

**Review by rmoyard — APPROVED** [review]
Thanks @soranjh, it looks good to me 💯

**Pull request merged**